coderd/workspaceagents.go

```diff
@@ -565,6 +576,18 @@ func (api *API) workspaceAgentLogs(rw http.ResponseWriter, r *http.Request) {
 				t.Reset(recheckInterval)
 			}
 
+			agents, err := api.Database.GetWorkspaceAgentsInLatestBuildByWorkspaceID(ctx, workspace.ID)
```

Comment:
What happens if the user wants to view logs for an old agent intentionally? I don't think we have any mechanism in the UI for this right now so it's probably fine to leave it like this

Reply:
It should just work, in this scenario essentially we (assuming `?follow=true`):

1. Fetch all logs in the first call to `GetWorkspaceAgentLogsAfter` (before websocket upgrade)
2. Websocket & write logs
3. Subscribe to db
4. Enter loop for buffered logs
5. keepGoing turns to false (non-latest build)
6. Check `GetWorkspaceAgentLogsAfter` one last time
7. Return